toolkit/authentication/src/androidTest/java/com/arcgismaps/toolkit/authentication/TestAuthenticationUtils.kt

```diff
@@ -0,0 +1,77 @@
+package com.arcgismaps.toolkit.authentication
+
+import android.content.Intent
+import android.net.Uri
+import com.arcgismaps.httpcore.ArcGISHttpClient
+import com.arcgismaps.httpcore.Request
+import com.arcgismaps.httpcore.Response
+import com.arcgismaps.httpcore.authentication.ArcGISAuthenticationChallenge
+import io.mockk.every
+import io.mockk.mockk
+
+/**
+ * A mock [ArcGISAuthenticationChallenge] for testing purposes.
+ *
+ * @since 200.5.0
+ */
+fun makeMockArcGISAuthenticationChallenge() = mockk<ArcGISAuthenticationChallenge>().apply {
+    every { requestUrl } returns "https://arcgis.com"
+    every { cause } returns Throwable()
+}
+
+/**
+ * Returns an intent that simulates a successful redirect from the OAuth server.
+ *
+ * @since 200.5.0
+ */
+fun getSuccessfulRedirectIntent(redirectUrl: String): Intent {
+    return Intent().apply {
+        data = Uri.parse("$redirectUrl?code=12345")
+        flags = Intent.FLAG_ACTIVITY_CLEAR_TOP
+    }
+}
+
+/**
+ * Intercepts requests to validate a token and returns a fake response that validates the token.
+ *
+ * @since 200.5.0
+ */
+fun ArcGISHttpClient.Builder.interceptTokenRequests() {
+    interceptor { chain ->
+        chain.request().let { request ->
+            if (request.url.endsWith("sharing/rest/oauth2/token")) {
+                request.getFakeTokenResponse()
+            } else chain.proceed(request)
+        }
+    }
+}
+
+/**
+ * Returns a fake response that validates the token.
+ *
+ * @since 200.5.0
+ */
+fun Request.getFakeTokenResponse(): Response =
```

Comment:
nitpick naming suggestion:
```suggestion
fun Request.createFakeTokenResponse(): Response =
```